wagtail_streamfield_migration_toolkit/management/commands/streamdatamigration.py

```diff
@@ -0,0 +1,230 @@
+from functools import lru_cache
+from django.core.management.base import BaseCommand, CommandError
+from django.db import connection
+from django.db.migrations.loader import MigrationLoader
+from django.db.migrations import Migration
+from django.db.migrations.autodetector import MigrationAutodetector
+from django.db.migrations.state import ProjectState
+from django.db.migrations.writer import MigrationWriter
+from django.apps import apps
+from wagtail.blocks import StreamBlock, StructBlock
+
+from wagtail_streamfield_migration_toolkit.migrate_operation import MigrateStreamData
+from wagtail_streamfield_migration_toolkit.operations import (
+    RemoveStreamChildrenOperation,
+    RenameStreamChildrenOperation,
+    RemoveStructChildrenOperation,
+    RenameStructChildrenOperation,
+)
+
+
+class Command(BaseCommand):
+
+    help = "Generate data migrations with the given operations. Note that you can only \
+        generate changes for models in a single app at a time."
+
+    def add_arguments(self, parser):
+        parser.add_argument("--name", help="Use this name for migration file")
+        subparsers = parser.add_subparsers(dest="operation_type")
+
+        rename_parser = subparsers.add_parser("rename", help="rename --help")
+        rename_parser.add_argument("app_label", help="app name")
+        rename_parser.add_argument("old_name", help="name of the block to be renamed")
+        rename_parser.add_argument("new_name", help="new name")
+        rename_parser.add_argument(
+            "paths",
+            nargs="+",
+            action="extend",
+            help="path/s to the block/s which is being operated on: '<model_name>.<field_name>.?<block_path>'",
+        )
+
+        remove_parser = subparsers.add_parser("remove", help="remove --help")
+        remove_parser.add_argument("app_label", help="app name")
+        remove_parser.add_argument("block_name", help="name of the block to remove")
+        remove_parser.add_argument(
+            "paths",
+            nargs="+",
+            action="extend",
+            help="path/s to the block/s which is being operated on: '<model_name>.<field_name>.?<block_path>'",
```

Comment:
Maybe we should include the app label here so you can operate across apps? `app_label.model_name` is a pretty standard way to reference a model in Django after all

Reply:
For now I was under the impression that this would work only for a single app at a time (single migration file). I wasn't sure if that wouldn't be clear when applabel is also part of the path. Otherwise yeah having it as part of the path would be more convenient too.

Reply:
Actually, that's a good point. In the future it would be nice to get it to generate multiple files in that case (one per app), but let's leave it as is for now then.